File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/TinTucColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TinTucRow } from "./TinTucRow";

export interface TinTucColumns {
    TinTucId: Column<TinTucRow>;
    DanhMucId: Column<TinTucRow>;
    TieuDe: Column<TinTucRow>;
    Slug: Column<TinTucRow>;
    TomTat: Column<TinTucRow>;
    AnhBia: Column<TinTucRow>;
    NoiDungHtml: Column<TinTucRow>;
    SeoTitle: Column<TinTucRow>;
    SeoDescription: Column<TinTucRow>;
    IsPublished: Column<TinTucRow>;
    IsHot: Column<TinTucRow>;
    ViewCount: Column<TinTucRow>;
    NgayDang: Column<TinTucRow>;
    NgayCapNhat: Column<TinTucRow>;
    TacGia: Column<TinTucRow>;
}

export class TinTucColumns extends ColumnsBase<TinTucRow> {
    static readonly columnsKey = 'SanPhamDB.TinTuc';
    static readonly Fields = fieldsProxy<TinTucColumns>();
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/TinTuc/TinTucColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.TinTuc")]
[BasedOnRow(typeof(TinTucRow), CheckNames = true)]
public class TinTucColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int TinTucId { get; set; }
    public int DanhMucId { get; set; }
    [EditLink]
    public string TieuDe { get; set; }
    public string Slug { get; set; }
    public string TomTat { get; set; }
    public string AnhBia { get; set; }
    public string NoiDungHtml { get; set; }
    public string SeoTitle { get; set; }
    public string SeoDescription { get; set; }
    public bool IsPublished { get; set; }
    public bool IsHot { get; set; }
    public int ViewCount { get; set; }
    public DateTime NgayDang { get; set; }
    public DateTime NgayCapNhat { get; set; }
    public string TacGia { get; set; }
}